build.gradle

```diff
@@ -143,15 +145,33 @@ task liquibaseHarnessTest(type: Test) {
 // Requires SPANNER_PROJECT and SPANNER_INSTANCE environment variables.
 // Also requires GOOGLE_APPLICATION_CREDENTIALS environment variable
 // if using a JSON key for authentication.
-task integrationTest(type: Test) {
-  shouldRunAfter "test"
+tasks.register('integrationTest', Test) {
+    shouldRunAfter "test"
 
-  // serviceLoaderBuild is necessary for Liquibase to find the extensions
-  dependsOn "serviceLoaderBuild"
+    // serviceLoaderBuild is necessary for Liquibase to find the extensions
+    dependsOn "serviceLoaderBuild"
 
-  useJUnitPlatform {
-    includeTags "integration"
-  }
+    useJUnitPlatform {
+        includeTags "integration"
+    }
+}
+
+tasks.withType(Test).configureEach {
+    javaLauncher = javaToolchains.launcherFor {
+        languageVersion = JavaLanguageVersion.of(8)
+    }
+    dependencies {
+        testImplementation("org.mockito:mockito-core:3.12.4")
+    }
+}
+
+tasks.named('liquibaseHarnessTest', Test).configure {
+    javaLauncher = javaToolchains.launcherFor {
+        languageVersion = JavaLanguageVersion.of(11)
+    }
+    dependencies {
+        testImplementation("org.mockito:mockito-core:5.14.1")
+    }
```

Comment:
nit: remove the empty `derby.log` file below (I can't add a comment to the empty file....)